opentracing-impl/src/test/java/io/opentracing/AbstractTracerTest.java

```diff
@@ -89,6 +89,23 @@ public void testNonEmptyExtract() {
         assertEquals("Should find the marker", "whatever", ((TestSpanBuilder)result).operationName);
     }
 
+    @Test
+    public void testExtractAsParent() throws Exception {
+        Map<String,String> map = Collections.singletonMap("test-marker", "whatever");
+        TextMapExtractAdapter adapter = new TextMapExtractAdapter(map);
+        AbstractTracer tracer = new TestTracerImpl();
+        SpanContext parent = tracer.extract(Format.Builtin.TEXT_MAP, adapter);
+        assert AbstractNoopSpan.INSTANCE != tracer.buildSpan("child").asChildOf(parent).start();
+    }
+
+    @Test
+    public void testExtractOfNoParent() throws Exception {
+        AbstractTracer tracer = new TestTracerImpl();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((Span)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf((SpanContext)AbstractNoopSpan.INSTANCE).start();
+        assert AbstractNoopSpan.INSTANCE == tracer.buildSpan("child").asChildOf(AbstractNoopSpanBuilder.INSTANCE).start();
```

Comment:
I agree about sampling, but I wasn't talking about sampling. At some point some node in the architecture is going to receive a request that has no trace ID from upstream. What I'm saying is this node should always start a trace in this case. If it doesn't do that, then a trace may never be started in the whole architecture. The trace may be sampled or not, that's up to the implementation of the tracer, but a new trace ID should be created at that point and propagated downstream. This is even more important from the point of view of distributed context propagation (baggage) than the tracing data itself.

The instrumentation does not decide on sampling, but it does decide on starting the trace by calling `tracer.startTrace()` - I don't see why this call should ever produce a noop span unless the app is configured with noop tracer.